snp_application/src/snp_widget.cpp

```diff
@@ -139,44 +139,51 @@ SNPWidget::SNPWidget(rclcpp::Node::SharedPtr rviz_node, QWidget* parent)
   board_->set("execute", static_cast<QAbstractButton*>(ui_->push_button_motion_execution));
 }
 
+BT::BehaviorTreeFactory SNPWidget::createBTFactory()
+{
+  BT::BehaviorTreeFactory bt_factory;
+
+  // Register custom nodes
+  bt_factory.registerNodeType<ButtonMonitorNode>("ButtonMonitor");
+  bt_factory.registerNodeType<ButtonApprovalNode>("ButtonApproval");
+  bt_factory.registerNodeType<ProgressDecoratorNode>("Progress");
+  bt_factory.registerNodeType<SetPageDecoratorNode>("SetPage");
+  bt_factory.registerNodeType<SNPSequenceWithMemory>("SNPSequenceWithMemory");
+  bt_factory.registerNodeType<RosSpinnerNode>("RosSpinner", bt_node_);
+
+  BT::RosNodeParams ros_params;
+  ros_params.nh = bt_node_;
+  ros_params.wait_for_server_timeout = std::chrono::seconds(0);
+  ros_params.server_timeout = std::chrono::seconds(get_parameter<int>(bt_node_, BT_SHORT_TIMEOUT_PARAM));
+
+  // Publishers/Subscribers
+  bt_factory.registerNodeType<ToolPathsPubNode>("ToolPathsPub", ros_params);
+  bt_factory.registerNodeType<MotionPlanPubNode>("MotionPlanPub", ros_params);
+  bt_factory.registerNodeType<UpdateTrajectoryStartStateNode>("UpdateTrajectoryStartState", ros_params);
+  // Short-running services
+  bt_factory.registerNodeType<TriggerServiceNode>("TriggerService", ros_params);
+  bt_factory.registerNodeType<GenerateToolPathsServiceNode>("GenerateToolPathsService", ros_params);
+  bt_factory.registerNodeType<StartReconstructionServiceNode>("StartReconstructionService", ros_params);
+  bt_factory.registerNodeType<StopReconstructionServiceNode>("StopReconstructionService", ros_params);
+
+  // Long-running services/actions
+  ros_params.server_timeout = std::chrono::seconds(get_parameter<int>(bt_node_, BT_LONG_TIMEOUT_PARAM));
```

Comment:
Maybe load the short and long timeout params outside of this method and pass in a `short_timeout_ros_params` and `long_timeout_ros_params` to this method? So the user doesn't have to get the parameters inside their new method.